### DISCUSSION TITLE
clickhouse function - time difference in hh:mm:ss format

### DISCUSSION THREAD

**Discussion by umatwav** [ClickHouse/ClickHouse#68946] [General]
Is there a ClickHouse function that outputs the difference between two DateTime values in HH:MM:SS format

**Comment by rschu1ze** [comment]
The difference between two `DateTime` values in ClickHouse is of type `Int` (it is not a `DateTime`).

If you like to format `DateTime`s, you can use function [formatDateTime](https://clickhouse.com/docs/en/sql-reference/functions/date-time-functions#formatdatetime).

**Reply by den-crane** [reply]
There is a function [formatReadableTimeDelta](https://clickhouse.com/docs/en/sql-reference/functions/other-functions#formatreadabletimedelta)

```
select formatReadableTimeDelta(5456456) r 
┌─r────────────────────────────────────────────────────┐
│ 2 months, 2 days, 3 hours, 40 minutes and 56 seconds │
└──────────────────────────────────────────────────────┘

SELECT formatReadableTimeDelta(5456456, 'hours') AS r
┌─r─────────────────────────────────────┐
│ 1515 hours, 40 minutes and 56 seconds │
└───────────────────────────────────────┘
```

>between two DateTime values in HH:MM:SS format

means this

```
select concat(intDiv(5456456 as d, 3600) as h,':', intDiv(d-h*3600, 60) as m, ':', d-h*3600-m*60) r;
┌─r──────────┐
│ 1515:40:56 │
└────────────┘
```
